master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,105 @@
+- If one or more network policies apply to a pod containing egress rules, then only the egress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+
+For other endpoint types (VMs, host interfaces), **Calico global network policy is default deny**. Only traffic specifically allowed by network policy is allowed, even if no network policies apply to the endpoint.
+
+#### Secure resources everywhere
+
+We recommend creating default deny policy using **Calico global network policy**. It applies to both workloads (VMs and containers) and hosts (computers that run the hypervisor for VMs, or container runtime for containers). Using a Calico global network policy supports a conservative security stance for protecting resources.
+
+#### Best practice: implicit default deny policy
+
+We recommend that you create a "plain vanilla" default deny Calico network policy that does not include other deny rules. Why? The absence of other deny rules ensures that the default deny policy is implicitly "first/always". 
+
+If you add other deny rules, you must also add an **order number** to ensure the policy always remains in affect (by virtue of being the highest order number across all of your ordered policies). An **implicit default deny policy** is not only easier, but avoids mistakes. 
+
+### How to
+
+Although you can use any of the following policies to create default deny for Kubernetes pods, we recommend using the Calico policy, non-namespaced.
+
+- [Create default deny-all traffic Calico policy, non-namespaced](#create-default-deny-all-traffic-calico-policy-non-namespaced)
+- [Create default deny-all traffic Calico policy, namespaced](#create-default-deny-all-traffic-calico-policy-namespaced)
+- [Create default deny all traffic Kubernetes policy, namespaced](#create-default-deny-all-traffic-Kubernetes-policy-namespaced)
+
+#### Create default deny-all traffic Calico policy, non-namespaced
+
+In the following example, we specify a default deny **GlobalNetworkPolicy** (recommended) that denies all traffic to workload endpoint resources in all namespaces, and to host endpoint resources. It is an implicit default deny policy without rules; it will always be in effect, regardless of the ordering scheme that you may specify in normal policies. 
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: default-deny
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+```
+
+#### Create default deny-all traffic Calico policy, namespaced  
+
+In the following example, we specify a default deny **NetworkPolicy** that denies all traffic for workloads in the namespace, **engineering**. It is an implicit default deny policy without rules; it will always be in effect, regardless of the ordering scheme that you may specify in normal policies. Note that the **selector: all()** is not required, but is used for clarity. 
```

Comment:
> Note that the **selector: all()** is not required, but is used for clarity. 

I say let's drop this - I think it is just confusing for the user.

Reply:
Fixed.